src/plugins/intel_cpu/src/nodes/executors/x64/subgraph.cpp

```diff
@@ -124,6 +124,67 @@ SubgraphExecutor::SubgraphExecutor(const std::shared_ptr<CPURuntimeConfig>& snip
+                                               const ov::intel_cpu::RepackedInput& repacked_input,
+                                               size_t tensor_rank) {
+    auto get_offset = [](const BlockedMemoryDescPtr& desc) {
+        return static_cast<ptrdiff_t>(desc->getOffsetPadding() * desc->getPrecision().size());
+    };
+
+    const auto* src_ptr =
+        srcMemPtr->getDataAs<const uint8_t>() + get_offset(srcMemPtr->getDescWithType<BlockedMemoryDesc>());
+    auto* dst_ptr = dstMemPtr->getDataAs<uint8_t>() + get_offset(dstMemPtr->getDescWithType<BlockedMemoryDesc>());
+
+    VectorDims dom;
+    const auto& shape = dstMemPtr->getShape().getDims();
+    OPENVINO_ASSERT(shape.size() <= tensor_rank, "Unsupported shape rank of repacking data");
+    init_parallel_domain(shape, tensor_rank, 2lu, dom);
+
+    const auto& in_strides = repacked_input.in_offsets();
+    const auto& out_strides = repacked_input.out_offsets();
+    OPENVINO_ASSERT(everyone_is(tensor_rank, in_strides.size(), out_strides.size(), dom.size()),
+                    "Unsupported shape rank of repacking data");
+
+    const auto& kernel = repacked_input.kernel<BrgemmCopyBKernel>();
+    if (tensor_rank == rank6D) {
+        parallel4d_repacking(kernel.get(), dom, in_strides, out_strides, src_ptr, dst_ptr);
+    } else {
+        parallelNd_repacking(kernel.get(), dom, in_strides, out_strides, src_ptr, dst_ptr);
+    }
+}
+
+std::vector<MemoryPtr> SubgraphExecutor::prepare_weights(const std::vector<MemoryPtr>& inMemPtrs,
+                                                         const RepackedInputConfig& repacked_const_input_config,
+                                                         const GraphContext::CPtr& context) {
+    std::vector<MemoryPtr> repackedMemPtrs = inMemPtrs;
+    for (const auto& p : repacked_const_input_config) {
+        const auto idx = p.first;
+        const auto& repacked_input = p.second;
+        OPENVINO_ASSERT(idx < inMemPtrs.size(), "Incorrect index of repacked input");
+        const auto& srcMemPtr = inMemPtrs[idx];
+
+        auto create = [&]() {
+            const auto& dstMemPtr = std::make_shared<Memory>(context->getEngine(), repacked_input.desc());
+            separately_repack_input(srcMemPtr, dstMemPtr, repacked_input, dstMemPtr->getShape().getDims().size());
+            return dstMemPtr;
+        };
+
+        auto weightCache = context->getWeightsCache();
+        if (weightCache != nullptr) {
+            const auto& wgtDims = srcMemPtr->getStaticDims();
+            OPENVINO_ASSERT(wgtDims.size() > 1, "Unexpected weight shape rank");
+            std::string format = "brgemm_snippets_" + std::to_string(wgtDims[0]) + "_" + std::to_string(wgtDims[1]);
+            const std::string string_hash = format + "_" + std::to_string(srcMemPtr->getSize()) + "_" +
+                                            std::to_string(reinterpret_cast<uint64_t>(srcMemPtr->getData()));
```

Comment:
It seems like we can easily reuse `DnnlExtensionUtils::computeWeightsStringHash` here to compute the hash. The only exclusive thing we need is unique snippets specific prefix ("brgemm_snippets_"). The rest can be reused

Reply:
Good catch. Thank you!